### PR TITLE
MasterRenderer refactor

### PR DESCRIPTION
**Pull request by est77**
- Remove unused BaseRenderer base class.
- Move implementation to private implementation.

### CODE REVIEW

File: src/appleseed/renderer/kernel/rendering/masterrenderer.cpp [hunk 1]
```diff
@@ -75,416 +80,563 @@ namespace renderer
...
+
+    // Initialize OSL's shading system.
+    bool initialize_osl_shading_system(
+        TextureStore&               texture_store,
+        foundation::IAbortSwitch&   abort_switch)
     {
-        Stopwatch<DefaultWallclockTimer> stopwatch;
-        stopwatch.start();
+        // Initialize OIIO.
+        const ParamArray& params = m_params.child("texture_store");
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/rendering/masterrenderer.cpp]: This variable is confusing. AFAICT, it's used only once so I would suggest inlining it.

File: src/appleseed/renderer/kernel/rendering/masterrenderer.cpp [hunk 2]
```diff
@@ -75,416 +80,563 @@ namespace renderer
...
+            {
+                for (size_t tx = 0; tx < frame_props.m_tile_count_x; ++tx)
+                    tile_callback->on_tile_end(frame, tx, ty);
+            }
+        }
     }
 
-    return true;
-}
+    Project&                    m_project;
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/rendering/masterrenderer.cpp]: Move data members to the top of the struct.

**Comment by est77** [comment]
Fixed and rebased

**Comment by dictoon** [comment]
Thanks!

**Pull request merged**